sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala

```diff
@@ -962,6 +963,22 @@ class QueryCompilationErrorsSuite
         "methodName" -> "update",
         "className" -> "org.apache.spark.sql.catalyst.expressions.UnsafeRow"))
   }
+
+  test("SPARK-42328: Convert unsupported data type from Spark to Parquet") {
+    val converter = new SparkToParquetSchemaConverter
+    val dummyDataType = new DataType {
+      override def defaultSize: Int = 0
+
+      override private[spark] def asNullable = NullType
+    }
+    checkError(
+      exception = intercept[AnalysisException] {
+        converter.convertField(StructField("test", dummyDataType))
```

Comment:
Is it possible to reproduce the error using public API. If not, we should think of converting to an internal error.

Reply:
@MaxGekk Replaced it with `INTERNAL_ERROR`, please check the updated PR description.